{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import paramiko\n",
    "import io"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# SSH connection details\n",
    "hostname = \"quest.northwestern.edu\"\n",
    "username = \"ekc5108\"\n",
    "password = \"MB~b1@Qt413%\"\n",
    "\n",
    "# Establish SSH connection\n",
    "client = paramiko.SSHClient()\n",
    "client.set_missing_host_key_policy(paramiko.AutoAddPolicy())\n",
    "client.connect(hostname, username=username, password=password)\n",
    "\n",
    "sftp = client.open_sftp()\n",
    "remote_file = sftp.open('/projects/b1044/Computational_Output/EKC/fmod_calc_runs_2/000000000524/035-EKC-A8C-75c-b-5-p_shapemapper_log.txt')\n",
    "# read txt file\n",
    "data = remote_file.read()\n",
    "remote_file.close()\n",
    "sftp.close()\n",
    "client.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "b\"\\n################################################################################\\nStarted ShapeMapper v2.2.0 at 2023-06-21 10:18:27\\nRunning from directory: /projects/b1044/Sequencing/SHAPE-Seq/EKC/20230617-DMS_fourU/035-EKC-A8C-75c-b-5-p\\nargs:  --name 035-EKC-A8C-75c-b-5-p --target ./fourU_A8C.fa --out 035-EKC-A8C-75c-b-5-p --modified --R1 ./035-EKC-A8C-75c-b-5-p_S35_L001_R1_001.fastq.gz --R2 ./035-EKC-A8C-75c-b-5-p_S35_L001_R2_001.fastq.gz --dms --amplicon --overwrite --nproc 24\\nUsing serial mode, which is currently required for DMS mode\\nUsing default DMS max-bg of 0.02\\n\\tYou can manually override using the max-bg flag\\nCreated pipeline at 2023-06-21 10:18:27\\nRunning PrimerLocator at 2023-06-21 10:18:27 . . .\\n. . . done at 2023-06-21 10:18:27\\nRunning FastaFormatChecker at 2023-06-21 10:18:27 . . .\\n. . . done at 2023-06-21 10:18:27\\nRunning BowtieIndexBuilder at 2023-06-21 10:18:27 . . .\\n. . . done at 2023-06-21 10:18:27\\nRunning ProgressMonitor at 2023-06-21 10:18:27 . . .\\n. . . done at 2023-06-21 10:18:28\\nRunning QualityTrimmer1 at 2023-06-21 10:18:28 . . .\\n. . . done at 2023-06-21 10:18:28\\nRunning QualityTrimmer2 at 2023-06-21 10:18:28 . . .\\n. . . done at 2023-06-21 10:18:29\\nRunning Interleaver at 2023-06-21 10:18:29 . . .\\n. . . done at 2023-06-21 10:18:30\\nRunning Merger at 2023-06-21 10:18:30 . . .\\n/```````````````````````````````````````````````````````````````````````````````\\n|Merger (sample: Modified) output message: \\n|----------------------------------------- \\n| \\n| java -Djava.library.path=/projects/b1044/Software/shapemapper2/internals/thirdparty/miniconda/envs/shapemapper_make/opt/bbmap-37.78/jni/ -ea -Xmx1000m -Xmx1000m -cp /projects/b1044/Software/shapemapper2/internals/thirdparty/miniconda/envs/shapemapper_make/opt/bbmap-37.78/current/ jgi.BBMerge vstrict=t in=stdin out=stdout outu=stdout interleaved=t usejni=t t=4\\n| Executing jgi.BBMerge [vstrict=t, in=stdin, out=stdout, outu=stdout, interleaved=t, usejni=t, t=4]\\n| Version 37.78 [vstrict=t, in=stdin, out=stdout, outu=stdout, interleaved=t, usejni=t, t=4]\\n| \\n| Revised arguments: [maxbad=4, margin=3, minqo=8, qualiters=2, ratiomode=t, flatmode=f, minentropy=52, minoverlap=12, minoverlap0=4, maxratio=0.05, ratiomargin=12, ratiooffset=0.5, ratiominoverlapreduction=4, efilter=2, pfilter=0.008, minsecondratio=0.16, minapproxoverlap=22, in=stdin, out=stdout, outu=stdout, interleaved=t, usejni=t, t=4]\\n| \\n| Set INTERLEAVED to true\\n| Set threads to 4\\n| Writing mergable reads merged.\\n| Unspecified format for output stdout; defaulting to fastq.\\n| Unspecified format for output stdout; defaulting to fastq.\\n| Started output threads.\\n| Unspecified format for input stdin; defaulting to fastq.\\n| Total time: 0.768 seconds.\\n| \\n| Pairs:               \\t233024\\n| Joined:              \\t36252   \\t15.557%\\n| Ambiguous:           \\t196772   \\t84.443%\\n| No Solution:         \\t0       \\t0.000%\\n| Too Short:           \\t0       \\t0.000%\\n| \\n| Avg Insert:          \\t106.0\\n| Standard Deviation:  \\t0.5\\n| Mode:                \\t106\\n| \\n| Insert range:        \\t50 - 112\\n| 90th percentile:     \\t106\\n| 75th percentile:     \\t106\\n| 50th percentile:     \\t106\\n| 25th percentile:     \\t106\\n| 10th percentile:     \\t106\\n\\\\_______________________________________________________________________________\\n. . . done at 2023-06-21 10:18:31\\nRunning Tab6Interleaver at 2023-06-21 10:18:31 . . .\\n. . . done at 2023-06-21 10:18:32\\nRunning BowtieAligner at 2023-06-21 10:18:32 . . .\\n/```````````````````````````````````````````````````````````````````````````````\\n|BowtieAligner (sample: Modified) output message: \\n|------------------------------------------------ \\n| \\n| 233024 reads; of these:\\n|   196772 (84.44%) were paired; of these:\\n|     180710 (91.84%) aligned concordantly 0 times\\n|     16061 (8.16%) aligned concordantly exactly 1 time\\n|     1 (0.00%) aligned concordantly >1 times\\n|     ----\\n|     180710 pairs aligned concordantly 0 times; of these:\\n|       0 (0.00%) aligned discordantly 1 time\\n|     ----\\n|     180710 pairs aligned 0 times concordantly or discordantly; of these:\\n|       361420 mates make up the pairs; of these:\\n|         184517 (51.05%) aligned 0 times\\n|         176897 (48.94%) aligned exactly 1 time\\n|         6 (0.00%) aligned >1 times\\n|   36252 (15.56%) were unpaired; of these:\\n|     146 (0.40%) aligned 0 times\\n|     36106 (99.60%) aligned exactly 1 time\\n|     0 (0.00%) aligned >1 times\\n| 57.03% overall alignment rate\\n\\\\_______________________________________________________________________________\\n. . . done at 2023-06-21 10:18:35\\nRunning MutationParser_Modified at 2023-06-21 10:18:35 . . .\\n. . . done at 2023-06-21 10:18:49\\nRunning MutationCounter_Modified at 2023-06-21 10:18:49 . . .\\n. . . done at 2023-06-21 10:18:51\\nRunning CalcProfile at 2023-06-21 10:18:51 . . .\\n. . . done at 2023-06-21 10:18:52\\nRunning NormProfile at 2023-06-21 10:18:52 . . .\\n/```````````````````````````````````````````````````````````````````````````````\\n|NormProfile (RNA: fourU_A8C_new) output message: \\n|------------------------------------------------ \\n| \\n| loaded profile and stderrs from shapemapper_temp/035-EKC-A8C-75c-b-5-p/PostAlignment_RNA_1/ProfileHandler/CalcProfile/035-EKC-A8C-75c-b-5-p_PostAlignment_RNA_1_ProfileHandler_CalcProfile_profile.txt\\n| (NormError(...), 'Error: sequence contains too few C nucleotides (=3) with quality reactivity information for effective per-nt normalization factor calculation')\\n\\\\_______________________________________________________________________________\\n. . . done at 2023-06-21 10:18:52\\nRunning TabToShape at 2023-06-21 10:18:52 . . .\\n. . . done at 2023-06-21 10:18:52\\nRunning RenderFigures at 2023-06-21 10:18:52 . . .\\n/```````````````````````````````````````````````````````````````````````````````\\n|RenderFigures (RNA: fourU_A8C_new) output message: \\n|-------------------------------------------------- \\n| \\n| \\n| DMS mode quality control checks:\\n| --------------------------------\\n| \\n| ***************\\n| *   Nt = A    *\\n| ***************\\n| \\n| Read depth check:\\n| 100.0% (12/12) nucleotides meet the minimum read depth of 5000\\n| PASS\\n| \\n| FAIL: Normalization unreliable due to low number of measured nucleotides!\\n|       Consider including another RNA as a normalization control\\n| \\n| ***************\\n| *   Nt = C    *\\n| ***************\\n| \\n| Read depth check:\\n| 100.0% (3/3) nucleotides meet the minimum read depth of 5000\\n| PASS\\n| \\n| FAIL: Normalization unreliable due to low number of measured nucleotides!\\n|       Consider including another RNA as a normalization control\\n| \\n| ***************\\n| *   Nt = U    *\\n| ***************\\n| \\n| Read depth check:\\n| 100.0% (13/13) nucleotides meet the minimum read depth of 5000\\n| PASS\\n| \\n| FAIL: Normalization unreliable due to low number of measured nucleotides!\\n|       Consider including another RNA as a normalization control\\n| \\n| ***************\\n| *   Nt = G    *\\n| ***************\\n| \\n| Read depth check:\\n| 100.0% (12/12) nucleotides meet the minimum read depth of 5000\\n| PASS\\n| \\n| FAIL: Normalization unreliable due to low number of measured nucleotides!\\n|       Consider including another RNA as a normalization control\\n| \\n| -----------------------------------------------------------------\\n|       A and/or C FAILURE: Possible causes:\\n|        - low sequencing depth\\n|        - DNA contamination\\n|        - poor mixing of chemical reagents and RNA and/or poor\\n|          reagent diffusion (if modifying in cells), resulting\\n|          in low modification rates\\n|        - expired reagents, resulting in low modification rates\\n|        - poor reverse transcription conditions, resulting in\\n|          low adduct read-through\\n|        - extremely highly structured RNA\\n| \\n\\\\_______________________________________________________________________________\\n. . . done at 2023-06-21 10:18:54\\nRunning RenderMappedDepths at 2023-06-21 10:18:54 . . .\\n/```````````````````````````````````````````````````````````````````````````````\\n|RenderMappedDepths (RNA: fourU_A8C_new) output message: \\n|------------------------------------------------------- \\n| \\n| Approximate maximum read depth for reads mapping near each amplicon primer pair\\n| (See *_mapped_depths.pdf for plots)\\n| (See *_per-amplicon_abundance.txt for exact numbers)\\n| \\n| Modified:\\n|   pair forward                       left    right   reverse                       left    right   depth  \\n|   ---- ----------------------------- ------- ------- ----------------------------- ------- ------- ------ \\n|   1    GGTGTAAGGGTGAAGTGTA           0       18      CTTCACCTTCATTACCTTTA          60      79      229k   \\n| \\n\\\\_______________________________________________________________________________\\n. . . done at 2023-06-21 10:18:55\\nShapeMapper run completed at 2023-06-21 10:18:55.\\nWARNING: This RNA has a possible poor quality reactivity profile: fourU_A8C_new\\nSee quality control checks above for details.\\n\""
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "data_analysis",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
