
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../..">
      
      
        <link rel="next" href="../nmr-kinetics/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.23">
    
    
      
        <title>Create Samples - nerd Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="light-blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#sample-organization-database-layout" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="nerd Documentation" class="md-header__button md-logo" aria-label="nerd Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            nerd Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Create Samples
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="light-blue"  aria-hidden="true"  type="radio" name="__palette" id="__palette_0">
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/edr-choi/nerd" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    nerd
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="./" class="md-tabs__link">
          
  
  
  Guides

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../cli/create/" class="md-tabs__link">
          
  
  
  CLI Reference

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="nerd Documentation" class="md-nav__button md-logo" aria-label="nerd Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    nerd Documentation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/edr-choi/nerd" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    nerd
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Create Samples
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Create Samples
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#schema-at-a-glance" class="md-nav__link">
    <span class="md-ellipsis">
      Schema at a Glance
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#core-task-tables" class="md-nav__link">
    <span class="md-ellipsis">
      Core Task Tables
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metadata-meta_" class="md-nav__link">
    <span class="md-ellipsis">
      Metadata (meta_…)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sample-creation-organization" class="md-nav__link">
    <span class="md-ellipsis">
      Sample Creation &amp; Organization
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mutational-counting-probe_fmod_" class="md-nav__link">
    <span class="md-ellipsis">
      Mutational Counting (probe_fmod_*)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#probe-time-course-fits-probe_tc_" class="md-nav__link">
    <span class="md-ellipsis">
      Probe Time-Course Fits (probe_tc_*)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nmr-experiments-nmr_" class="md-nav__link">
    <span class="md-ellipsis">
      NMR Experiments (nmr_*)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#temperature-gradient-fits-tempgrad_" class="md-nav__link">
    <span class="md-ellipsis">
      Temperature-Gradient Fits (tempgrad_*)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#navigating-the-database" class="md-nav__link">
    <span class="md-ellipsis">
      Navigating the Database
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../nmr-kinetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NMR Kinetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../probe-mutcount/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Modification Counting
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../probe-timecourse/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Probe Timecourse
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tempgrad-fit/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Temperature Gradients
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    CLI Reference
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            CLI Reference
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cli/create/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    create
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cli/nmr_create/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    nmr_create
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cli/nmr_deg_kinetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    nmr_deg_kinetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cli/nmr_add_kinetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    nmr_add_kinetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cli/mut_count/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    mut_count
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cli/probe_timecourse/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    probe_timecourse
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cli/tempgrad_fit/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    tempgrad_fit
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#schema-at-a-glance" class="md-nav__link">
    <span class="md-ellipsis">
      Schema at a Glance
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#core-task-tables" class="md-nav__link">
    <span class="md-ellipsis">
      Core Task Tables
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metadata-meta_" class="md-nav__link">
    <span class="md-ellipsis">
      Metadata (meta_…)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sample-creation-organization" class="md-nav__link">
    <span class="md-ellipsis">
      Sample Creation &amp; Organization
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mutational-counting-probe_fmod_" class="md-nav__link">
    <span class="md-ellipsis">
      Mutational Counting (probe_fmod_*)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#probe-time-course-fits-probe_tc_" class="md-nav__link">
    <span class="md-ellipsis">
      Probe Time-Course Fits (probe_tc_*)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nmr-experiments-nmr_" class="md-nav__link">
    <span class="md-ellipsis">
      NMR Experiments (nmr_*)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#temperature-gradient-fits-tempgrad_" class="md-nav__link">
    <span class="md-ellipsis">
      Temperature-Gradient Fits (tempgrad_*)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#navigating-the-database" class="md-nav__link">
    <span class="md-ellipsis">
      Navigating the Database
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="sample-organization-database-layout">Sample Organization &amp; Database Layout</h1>
<p>NERD persists every piece of metadata, analysis configuration, and fit result inside a single SQLite database (<code>nerd.sqlite</code>). A lightweight relational store gives us a few big wins:</p>
<ul>
<li>Transactions keep inserts and updates consistent, even when multiple CLI tasks run in sequence.</li>
<li>Foreign keys model the lab reality—constructs, buffers, reaction groups, and fit runs link explicitly.</li>
<li>The file is portable: copy <code>nerd.sqlite</code> alongside figures and configs to reproduce an analysis.</li>
</ul>
<p>Think of the database as the canonical record of an experiment. CSVs define starting metadata, FASTQs supply raw reads, fits generate parameters, but the SQLite file ties everything together.</p>
<hr />
<h2 id="schema-at-a-glance">Schema at a Glance</h2>
<p>Tables are grouped by function:</p>
<ul>
<li><strong>core_</strong>* – task orchestration (<code>core_tasks</code>, <code>core_task_members</code>).</li>
<li><strong>meta_</strong>* – reusable metadata (constructs, buffers, nucleotides).</li>
<li><strong>sample creation</strong> – sequencing runs, samples, reaction groups, and probe reactions.</li>
<li><strong>probe_fmod*</strong> – outputs from mutational counting pipelines (ShapeMapper, etc.).</li>
<li><strong>probe_tc*</strong> – free/global/constrained time-course fit runs and parameters.</li>
<li><strong>nmr_</strong>* – NMR reaction definitions, trace files, and kinetics fits.</li>
<li><strong>tempgrad_</strong>* – Arrhenius and melt fits derived from probe or NMR data.</li>
</ul>
<p>The sections below sketch how each area fits together so you can quickly orient in the database browser of your choice.</p>
<hr />
<h2 id="core-task-tables">Core Task Tables</h2>
<table>
<thead>
<tr>
<th>Table</th>
<th>Purpose</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>core_tasks</code></td>
<td>One row per CLI task execution (e.g., <code>create</code>, <code>mut_count</code>). Stores status, config hash, timestamps.</td>
</tr>
<tr>
<td><code>core_task_members</code></td>
<td>Links a task to specific entities (sample IDs, reaction groups) to capture scope.</td>
</tr>
<tr>
<td><code>core_cached_tasks</code></td>
<td>Records when a task reuses prior results via the caching mechanism.</td>
</tr>
</tbody>
</table>
<p>These tables are the audit trail. If the CLI reports that a run completed, you’ll see a matching row with <code>state='completed'</code>. Failures, reruns, and cached executions are all tracked here.</p>
<hr />
<h2 id="metadata-meta_">Metadata (<code>meta_…</code>)</h2>
<table>
<thead>
<tr>
<th>Table</th>
<th>Key Columns</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>meta_constructs</code></td>
<td><code>disp_name</code>, <code>family</code>, <code>version</code></td>
<td>Unique construct definitions.</td>
</tr>
<tr>
<td><code>meta_nucleotides</code></td>
<td><code>construct_id</code>, <code>site</code>, <code>base</code></td>
<td>Per-nucleotide metadata; generated if not provided.</td>
</tr>
<tr>
<td><code>meta_buffers</code></td>
<td><code>name</code>, <code>pH</code>, <code>composition</code></td>
<td>Buffer recipes referenced by reactions.</td>
</tr>
</tbody>
</table>
<p>Constructs and buffers are shared resources. When you import new samples, the CLI resolves construct/display names back into these tables. Deleting a construct cascades to its nucleotides thanks to foreign keys.</p>
<hr />
<h2 id="sample-creation-organization">Sample Creation &amp; Organization</h2>
<table>
<thead>
<tr>
<th>Table</th>
<th>Purpose</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>sequencing_runs</code></td>
<td>One row per instrument run (MiSeq, NovaSeq, etc.).</td>
</tr>
<tr>
<td><code>sequencing_samples</code></td>
<td>Parent samples pointing to FASTQ directories/files.</td>
</tr>
<tr>
<td><code>sequencing_derived_samples</code></td>
<td>Virtual samples produced by subsampling/filtering operations.</td>
</tr>
<tr>
<td><code>probe_reaction_groups</code></td>
<td>Labels that group related probe reactions (e.g., time course replicates).</td>
</tr>
<tr>
<td><code>probe_reactions</code></td>
<td>Connects a sequencing sample to constructs, buffers, probe chemistry, and reaction conditions.</td>
</tr>
</tbody>
</table>
<p>Typical flow: <code>create</code> inserts sequencing runs and samples, <code>probe_reaction_groups</code> tracks the lab-defined grouping (e.g., <code>65_1</code>), and <code>probe_reactions</code> attaches experimental context (temperature, probe concentration). Derived samples record how a <code>mut_count</code> task generated filtered FASTQs; parent-child relationships let downstream tasks trace provenance.</p>
<p>Foreign-key highlights:</p>
<ul>
<li><code>probe_reactions.s_id → sequencing_samples.id</code></li>
<li><code>probe_reactions.construct_id → meta_constructs.id</code></li>
<li><code>probe_reactions.buffer_id → meta_buffers.id</code></li>
</ul>
<hr />
<h2 id="mutational-counting-probe_fmod_">Mutational Counting (<code>probe_fmod_*</code>)</h2>
<table>
<thead>
<tr>
<th>Table</th>
<th>Purpose</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>probe_fmod_runs</code></td>
<td>Each invocation of a mutational counting tool for a sample.</td>
</tr>
<tr>
<td><code>probe_fmod_values</code></td>
<td>Per-nucleotide reactivity (e.g., <code>modrate</code>) plus depth and QC flags.</td>
</tr>
</tbody>
</table>
<p>When you run <code>nerd run mut_count</code>, the task registers a <code>probe_fmod_runs</code> row, inserts per-nucleotide values, and marks outliers. <code>probe_fmod_values.nt_id</code> points back to <code>meta_nucleotides</code>, so all reactivity data stays linked to constructs.</p>
<hr />
<h2 id="probe-time-course-fits-probe_tc_">Probe Time-Course Fits (<code>probe_tc_*</code>)</h2>
<table>
<thead>
<tr>
<th>Table</th>
<th>Purpose</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>probe_tc_fit_runs</code></td>
<td>Catalog of free/global/constrained fits. Captures round type, reaction group, valtype, and optional fmod run linkage.</td>
</tr>
<tr>
<td><code>probe_tc_fit_params</code></td>
<td>Tall table storing fit parameters, errors, or QC metrics keyed by <code>fit_run_id</code>.</td>
</tr>
</tbody>
</table>
<p>Each time-course task can spawn multiple rounds; the CLI writes one <code>probe_tc_fit_runs</code> row per round or scope. Parameters (e.g., <code>log_kobs</code>, <code>log_kdeg</code>) are stored as rows in <code>probe_tc_fit_params</code>, making it easy to extend with new metrics.</p>
<hr />
<h2 id="nmr-experiments-nmr_">NMR Experiments (<code>nmr_*</code>)</h2>
<table>
<thead>
<tr>
<th>Table</th>
<th>Purpose</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>nmr_reactions</code></td>
<td>Experiment definition (reaction type, substrate, buffer, replicate info).</td>
</tr>
<tr>
<td><code>nmr_trace_files</code></td>
<td>File registry per reaction and role (decay trace, peak trace, DMS trace).</td>
</tr>
<tr>
<td><code>nmr_fit_runs</code></td>
<td>Each kinetic model execution, including plugin name and status.</td>
</tr>
<tr>
<td><code>nmr_fit_params</code></td>
<td>Tall parameter storage for kinetics outputs (<code>k_value</code>, <code>k_error</code>, R², etc.).</td>
</tr>
</tbody>
</table>
<p>The workflow:</p>
<ol>
<li><code>nmr_create</code> registers trace files and reaction metadata.</li>
<li><code>nmr_deg_kinetics</code> / <code>nmr_add_kinetics</code> draw inputs, stage trace files into a run directory, and execute the named plugin.</li>
<li>Results land in <code>nmr_fit_runs</code> (one row per reaction fit) with parameters normalized into <code>nmr_fit_params</code>.</li>
</ol>
<hr />
<h2 id="temperature-gradient-fits-tempgrad_">Temperature-Gradient Fits (<code>tempgrad_*</code>)</h2>
<table>
<thead>
<tr>
<th>Table</th>
<th>Purpose</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>tempgrad_fit_runs</code></td>
<td>Top-level record for each Arrhenius or two-state fit execution.</td>
</tr>
<tr>
<td><code>tempgrad_series</code></td>
<td>Series metadata (construct, buffer, probe) for individual fit curves.</td>
</tr>
<tr>
<td><code>tempgrad_series_params</code></td>
<td>Parameters per series (activation energy, slopes, intercepts).</td>
</tr>
<tr>
<td><code>tempgrad_series_diagnostics</code></td>
<td>Diagnostics like R², RMSE, weight usage.</td>
</tr>
</tbody>
</table>
<p>Arrhenius runs use per-series regression; two-state fits may share baselines and store additional free energy estimates. Every series references the reaction group or source data so you can trace the input pipeline.</p>
<hr />
<h2 id="navigating-the-database">Navigating the Database</h2>
<p>Open <code>nerd.sqlite</code> in your favorite viewer (DB Browser for SQLite, Datasette, TablePlus). A few orientation tips:</p>
<ul>
<li><strong>Start with <code>core_tasks</code></strong> to see what ran. Use <code>task_id</code> to jump to downstream artifacts (<code>nmr_fit_runs.task_id</code>, provenance logs under the run directory).</li>
<li><strong>Follow foreign keys</strong>. Most tables use integer IDs with descriptive columns (<code>rg_label</code>, <code>sample_name</code>) for human readability.</li>
<li><strong>Tall parameter tables</strong> (<code>*_fit_params</code>) include both numeric and text columns. Filter by <code>param_name</code> to extract specific values.</li>
<li><strong>JSON artifacts</strong>. Tasks also dump companion JSON files in <code>run/output_dir/&lt;label&gt;/&lt;task&gt;/create___cfg-.../results</code>. The SQLite entries are designed to point back to those files via paths stored in metadata tables.</li>
</ul>
<p>With this structure your analyses are reproducible: rerunning a task updates entries, derived tables stay in sync, and every parameter ties back to the sample and construct that produced it. Use the CLI to orchestrate; use the database to audit, explore, and share.</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tabs", "navigation.sections", "content.code.copy"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
    
  </body>
</html>